components/frontend/src/widgets/toast.js

```diff
@@ -1,8 +1,12 @@
 import { toast } from 'react-toastify';
+import { registeredURLSearchParams } from '../utils'
 
 export function showMessage(type, title, description) {
-    const toast_message = title && description ? <><h4>{title}</h4><p>{description}</p></> : title;
-    toast(toast_message, { type: type, autoClose: 20000 });
+    const hideToasts = registeredURLSearchParams().get("hide_toasts");
```

Comment:
Is it not a bit weird to infer the frontend logic from the API call it made itself, or am I reading this incorrectly?

Reply:
That's because there are two frontends running: one in the user's browser and one in the headless browser of the renderer. When the user clicks "export report to PDF" the renderer opens the report, that the user has open in their browser, in a headless browser. The report when opened in the headless browser should not display toasts because we do not want them to end up in the PDF.